File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        int n = arr.size();
        int chunk = 0, prefixSum = 0, sortedPrefixSum = 0;
        for (int i = 0; i < n; i++) {
            prefixSum += arr[i];
            sortedPrefixSum += i;
            if (sortedPrefixSum == prefixSum) {
                chunk++;
            }
        }
        return chunk;
    }
};

File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        n = len(arr)
        chunk = 0
        prefixSum = 0
        sortedPrefixSum = 0
        for i in range(n):
            prefixSum += arr[i]
            sortedPrefixSum += i
            if sortedPrefixSum == prefixSum:
                chunk += 1
        return chunk